x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java

```diff
@@ -100,6 +118,13 @@ boolean isSnapshot() {
         return Build.CURRENT.isSnapshot();
     }
 
+    @Override
+    public Collection<Object> createComponents(Client client, ClusterService clusterService, ThreadPool threadPool, ResourceWatcherService resourceWatcherService, ScriptService scriptService, NamedXContentRegistry xContentRegistry, Environment environment, NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry, IndexNameExpressionResolver indexNameExpressionResolver, Supplier<RepositoriesService> repositoriesServiceSupplier) {
+        Set<AutoscalingDeciderService<?>> services =
+            autoscalingPlugins.stream().flatMap(p -> p.deciders().stream()).collect(Collectors.toSet());
```

Comment:
`Autoscaling` can immediately create the `AutoscalingDeciderServices` object containing the full list of deciders, including all extension's deciders.